src/ui.jsx

```diff
@@ -0,0 +1,392 @@
+/**
+ * ui.jsx
+ *
+ * A file describing the UI of the application using React. A lot of the state accesses occur on the state property
+ * which apparently isn't the best practice but it gets rid of lots of unnecessary re-renders.
+ *
+ */
+"use strict";
+/*eslint-disable no-console, no-underscore-dangle*/
```

Comment:
I've thrown these into the package.json with the rest of the ESLint definitions, should I move them into an .eslintrc?